pypmc/sampler/markov_chain.py

```diff
@@ -132,27 +137,31 @@ def run(self, N=1):
             log_rho = get_log_rho(proposed_point, proposed_eval)
 
             # check for NaN
-            if _np.isnan(log_rho): raise ValueError('encountered NaN')
-
-            # accept if rho = 1
-            if log_rho >=0:
-                accept_count += 1
-                this_run[i_N]            = proposed_point
-                self.current_point       = proposed_point
-                self.current_target_eval = proposed_eval
-
-            # accept with probability rho
-            elif log_rho >= _np.log(self.rng.rand()):
-                accept_count += 1
-                this_run[i_N]            = proposed_point
-                self.current_point       = proposed_point
-                self.current_target_eval = proposed_eval
-
-            # reject if not accepted
-            else:
+            if _np.isnan(log_rho):
+                # raise error if so desired
+                if not continue_on_NaN: raise ValueError('encountered NaN')
+                # otherwise reject
                 this_run[i_N] = self.current_point
-                #do not need to update self.current
-                #self.current = self.current
+            else:
+                # accept if rho = 1
+                if log_rho >=0:
```

Comment:
How could I allow this code duplication? All of this should disappear and we can just check for `log_rho >= _np.log(self.rng.rand())`

Reply:
I would prefer if we deduplicated code in a separate commit, or not at all. It is very readable in this way. But it's your decision in the end!

Reply:
Also, it comes to mind that your proposed condition changes the number of evaluations of ```rng```. Would make old computation non-repeatable. This is not the case for my changes, since previous computations would simply fail.

Reply:
Ok, I vaguely remember we used to do something different in the two branches but we don't anymore. Yep, it's a different issue. So let's leave as is. And good point about rng evaluation. Easily solved with short-circuit semantics
```python
if log_rho >=0 or log_rho >= _np.log(self.rng.rand()):
```